python/ctsm/crop_calendars/regrid_ggcmi_shdates.py

```diff
@@ -32,47 +33,106 @@ def define_arguments(parser):
     parser.add_argument(
         "-rt",
         "--regrid-template-file",
-        help="Template file to be used in regridding of inputs.",
+        help="Template netCDF file to be used in regridding of inputs. This can be a CLM output file (i.e., something with 1-d lat and lon variables) or a CLM surface dataset (i.e., something with 2-d LATIXY and LONGXY variables).",
         type=str,
         required=True,
     )
     return parser
 
 
+# lat or lon
+def import_coord_1d(ds, coordName):
+    da = ds[coordName]
+    if len(da.dims) != 1:
+        raise RuntimeError(f"Expected 1 dimension for {coordName}; found {len(da.dims)}: {da.dims}")
+    return da, len(da)
+
+
+# LATIXY or LONGXY
+def import_coord_2d(ds, coordName, varName):
+    da = ds[varName]
+    thisDim = [x for x in da.dims if coordName in x]
+    if len(thisDim) != 1:
+        raise RuntimeError(
+            f"Expected 1 dimension name containing {coordName}; found {len(otherDim)}: {otherDim}"
+        )
+    thisDim = thisDim[0]
+    otherDim = [x for x in da.dims if coordName not in x]
+    if len(otherDim) != 1:
+        raise RuntimeError(
```

Comment:
Something that could use a unit test for a error check.

Reply:
Done in 96b88aa.